dcp_prototype/backend/wrangling/migrations/scripts/migrate_dcp_one_to_dcp_two.py

```diff
@@ -19,27 +17,27 @@
 import queue
 import threading
 import time
-
-BUCKET_NAME = "hca-dcp-one-backup-data"
-PREFIX = "single_cell_transcriptome_analysis_of_human_pancreas_metadata"
-
-S3_CLIENT = boto3.client("s3")
-
-
-def generate_metadata_tsv_name(project_name):
-    """
-    Given a project name, generates a name for the spreadsheet that will be generated
-    based on this project's metadata.
-    """
-    replace_spaces = project_name.replace(" ", "_")
-    replace_slashes = replace_spaces.replace("/", "_")
-    return replace_slashes + "_metadata.xlsx"
-
-
-def get_entity_type(filename):
-    for key in ENTITY_TYPES:
-        if key in filename:
-            return key
+import tarfile
+
+
+def get_entity_type(object_name, object_data):
+    desc = object_data.get("describedBy")
+    if desc:
+        path = urlparse(desc).path
+        entity_type = os.path.basename(path)
+        if entity_type not in ENTITY_TYPES:
+            if entity_type in (
```

Comment:
Wouldn't this be redundant to the previous conditional?

Reply:
The purpose is to make it easy to see what entity types we are ignoring, and being explicit about that.   We are currently ignoring the 5 that are listed.  If we see anything else that is not in ENTITY_TYPES then we raise an exception.

Reply:
Oh, it's not redundant logic.  The 5 listed are not part if ENTITY_TYPES.

Reply:
Got it, thanks.